diesel/src/pg/types/date_and_time/mod.rs

```diff
@@ -140,12 +140,19 @@ impl ToSql<sql_types::Interval, Pg> for PgInterval {
 }
 
 impl FromSql<sql_types::Interval, Pg> for PgInterval {
-    fn from_sql(bytes: Option<&[u8]>) -> deserialize::Result<Self> {
-        let bytes = not_none!(bytes);
+    fn from_sql(value: Option<PgValue>) -> deserialize::Result<Self> {
+        let value = not_none!(value);
+        let bytes = value.bytes();
         Ok(PgInterval {
-            microseconds: FromSql::<sql_types::BigInt, Pg>::from_sql(Some(&bytes[..8]))?,
-            days: FromSql::<sql_types::Integer, Pg>::from_sql(Some(&bytes[8..12]))?,
-            months: FromSql::<sql_types::Integer, Pg>::from_sql(Some(&bytes[12..16]))?,
+            microseconds: FromSql::<sql_types::BigInt, Pg>::from_sql(Some(
+                PgValue::with_oid(&bytes[..8], None) // TODO Find OID
```

Comment:
We should set an oid here. Basically we need to find a way to call `HasSqlType::metadata` for types with a static oid.

Reply:
@weiznich It looks like the impl for PgMetadataLoop and PgTypeMetadata both query pg_type, which wouldn't be performant. What does the PGftype call in pg/connection/result.rs do that requires us to redefine the OID later?